src/main/java/org/candlepin/resource/OwnerResource.java

```diff
@@ -1239,8 +1240,14 @@ public Stream<PoolDTO> listOwnerPools(
 
         // Process the filters passed for the attributes
         PoolFilterBuilder poolFilters = new PoolFilterBuilder();
-        for (KeyValueParamDTO filterParam : attrFilters) {
-            poolFilters.addAttributeFilter(filterParam.getKey(), filterParam.getValue());
+        for (String filterParam : attrFilters) {
+            String[] keyValue = filterParam.split(":");
+            if (keyValue.length == 2) {
```

Comment:
This check doesn't line up with how we perform this processing in other places, even within the same file, nor does it match with the previous behavior. Previous builds split the key:value on the first instance of a colon, with all others being part of the value.

Whenever we make API changes like this, we need to ensure that Katello and Teamnado, at least, will not be negatively affected by such a change. Otherwise we need to keep the behavior the same. Additionally, we want to be consistent in how we accept these things, as it's very disruptive and confusing to users if a format is acceptable in one place, not in another, behaves differently in a third, and so on.

Reply:
Right, the main difference here is the original way of parsing it left colons in the value alone. That is, the string `"key:value:with:colons"` would be parsed to:
- key: "key"
- value: "value:with:colons"

The parsing code in this PR has a few different methods. One will split the string into "key" and "value" and then silently discard the rest. Others will split into those two strings and then throw an error because of the extra colons. Neither of which lines up with the old behavior of only splitting on the first colon and then only throwing an exception if there was nothing to split on at all.

In the name of standardizing all of this: do we have a way to use the old converter again, such that each time we receive a key-value parameter we aren't manually parsing it ourselves? If not, then let's add a utility function for doing this parsing one or more key-value pairs so we have a centralized place that is guaranteed to be following the same rules.

As far as the translation goes, I was surprised by the CandlepinParameterParseException because that is *also* not at all how we're supposed to generate those, but since that's how we've done it in the past, that's acceptable here.

Reply:
i create class with old parsing so it should be working now.